### PR TITLE
Incorporate comments from Stuart's review

### PR DESCRIPTION
**Pull request by tfpauly**
This covers several comments that @StuartCheshire had on the documents!

### CODE REVIEW

File: draft-pauly-adaptive-dns-privacy.md [hunk 1]
```diff
@@ -261,6 +260,12 @@ expected to support acting as a target for Oblivious DoH. A client MUST issue at
 least one query that is targeted at the server through a proxy before sending direct queries
 to the server.
 
+Designated DoH Servers are expected to act both as Oblivious Proxies and Oblivious Targets
```
Comment (erickinnear) [inline draft-pauly-adaptive-dns-privacy.md]:
```suggestion
Designated DoH Servers are expected to act both as Oblivious Proxies and as Oblivious Targets
```

File: draft-pauly-adaptive-dns-privacy.md [hunk 2]
```diff
@@ -275,8 +280,10 @@ as ".com".
 When a Designated DoH Server is discovered, clients SHOULD also check to see
 if this server provides an extended configuration in the form of a Web PvD ({{configuration}}).
 To do this, the client performs a GET request to the DoH URI, indicating that it accepts
-a media type of “application/pvd+json” ({{!I-D.ietf-intarea-provisioning-domains}})
-and with empty query and fragment components.
+a media type of “application/pvd+json” {{!I-D.ietf-intarea-provisioning-domains}}. When requesting
+the PvD information, the query and fragment components of the requested path SHOULD be left
+empty. Note that this is different from GET requests for the “application/dns-message” type,
```
Comment (erickinnear) [inline draft-pauly-adaptive-dns-privacy.md]:
```suggestion
empty. Note that this is different from a GET request for the “application/dns-message” type,
```

File: draft-pauly-adaptive-dns-privacy.md [hunk 3]
```diff
@@ -334,11 +341,20 @@ See {{local-deployment}} for local deployment considerations.
 
 When establishing a secure connection to a certain hostname, clients need
 to first determine which resolver configuration ought to be used for DNS resolution.
+
+Several of the steps outlined in this algorithm take into account the success or failure
+of name resolution. Failure can be indicated either by a DNS response, such as SERVFAIL
+NXDOMAIN, or by a connection-level failure, such as a TCP reset, and TLS handshake failure,
```
Comment (erickinnear) [inline draft-pauly-adaptive-dns-privacy.md]:
```suggestion
or NXDOMAIN, or by a connection-level failure, such as a TCP reset, and TLS handshake failure,
```

File: draft-pauly-adaptive-dns-privacy.md [hunk 4]
```diff
@@ -702,29 +721,27 @@ on such networks to prevent malicious interception.
...
+One of the challenges of using non-local DNS servers (such as cloud-based DoH servers)
+is that recursive queries made by these servers will originate from an IP address that
+is not necessarily geographically related to the client. Many DNS servers make assumptions
+about the geographic locality of clients to their recursive resolvers to optimize answers.
+To avoid this problem, the client's subnet can be forwarded to the authoritative server
+by the recursive using the EDNS0 Client Subnet feature. Oblvious DoH discourages this practice
+for privacy reasons. However, sharing this subnet, while detrimental to privacy, can result in
+better targeted DNS resolutions.
+
+Adaptive DNS splits DoH queries up into two sets: those made to Designated DoH Servers,
```
Comment (erickinnear) [inline draft-pauly-adaptive-dns-privacy.md]:
```suggestion
Adaptive DNS splits DoH queries into two sets: those made to Designated DoH Servers,
```

File: draft-pauly-adaptive-dns-privacy.md [hunk 5]
```diff
@@ -261,6 +260,12 @@ expected to support acting as a target for Oblivious DoH. A client MUST issue at
 least one query that is targeted at the server through a proxy before sending direct queries
 to the server.
 
+Designated DoH Servers are expected to act both as Oblivious Proxies and as Oblivious Targets
+to ensure that clients have sufficient options for preserving privacy using Oblivious DoH.
+Oblivious Targets are expected to act as Oblivious Proxies to ensure that no Oblivious DoH server
+can act as only a target (thus being able to see patterns in name resolution, which might have
+value to a resolver) and require other servers to take on a disproportionate load of proxying.
```
Comment (chris-wood) [inline draft-pauly-adaptive-dns-privacy.md]: We should probably also recommend some client behavior if certain targets are not willing to proxy. Perhaps, "Clients SHOULD stop using Oblivious Targets in the absence of proxy support"?

File: draft-pauly-adaptive-dns-privacy.md [hunk 6]
```diff
@@ -334,11 +341,20 @@ See {{local-deployment}} for local deployment considerations.
 
 When establishing a secure connection to a certain hostname, clients need
 to first determine which resolver configuration ought to be used for DNS resolution.
+
+Several of the steps outlined in this algorithm take into account the success or failure
+of name resolution. Failure can be indicated either by a DNS response, such as SERVFAIL
+or NXDOMAIN, or by a connection-level failure, such as a TCP reset, and TLS handshake failure,
```
Comment (chris-wood) [inline draft-pauly-adaptive-dns-privacy.md]:
```suggestion
or NXDOMAIN, or by a connection-level failure, such as a TCP reset, TLS handshake failure,
```

File: draft-pauly-oblivious-doh.md [hunk 7]
```diff
@@ -106,8 +106,12 @@ Oblivious DoH requires, at a minimum:
 - Two DoH servers, where one can act as an Oblivious Proxy, and the other can act as an
 Oblivious Target.
 - Public keys for encrypting DNS queries that are passed from a client through a proxy
-to a target ({{publickey}}).
-- Client ability to generate one-time-use symmetric keys to encrypt DNS responses.
+to a target ({{publickey}}). These keys guarantee that only the intended Oblivious Target
+can decrypt client queries.
+- Client ability to generate one-time-use symmetric keys to encrypt DNS responses. These
```
Comment (chris-wood) [inline draft-pauly-oblivious-doh.md]:
```suggestion
- Client ability to generate random {{!RFC4086}} one-time-use symmetric keys to encrypt DNS responses. These
```

File: draft-pauly-adaptive-dns-privacy.md [hunk 8]
```diff
@@ -229,12 +229,11 @@ the HTTPSSVC variant of the SVCB record type on "example.com".
...
 
-When a client resolves a name (based on the order in
-{{resolution-algorithm}}) it SHOULD determine the Designated DoH
-Server via a SVCB record for any name that does not fall within known
-Designated DoH Server's configuration. The client MAY also issue
-queries for the SVCB record for more specific names to discover
-further Designated DoH Servers.
+Whenever a client resolves a name for which it does not already have a Designated DoH Server,
+it SHOULD try to determine the Designated DoH Server by sending a query for the appropriate
+SVCB record. If there is no DoH server designated for the name or zone, signalled either
```
[the remark this answers is not in the document]
Comment (tfpauly) [reply draft-pauly-adaptive-dns-privacy.md]: Changed the wording to just say its for the same name!

File: draft-pauly-adaptive-dns-privacy.md (hunk 5 above)
Comment (tfpauly) [reply draft-pauly-adaptive-dns-privacy.md]: I think this is already covered by the beginning of the section, where the clients MUST whitelist the servers by validating that they can both be the proxy and target. You should never use a server as a target if it isn't a proxy too.

File: draft-pauly-adaptive-dns-privacy.md [hunk 9]
```diff
@@ -275,8 +280,10 @@ as ".com".
 When a Designated DoH Server is discovered, clients SHOULD also check to see
 if this server provides an extended configuration in the form of a Web PvD ({{configuration}}).
 To do this, the client performs a GET request to the DoH URI, indicating that it accepts
-a media type of “application/pvd+json” ({{!I-D.ietf-intarea-provisioning-domains}})
-and with empty query and fragment components.
+a media type of “application/pvd+json” {{!I-D.ietf-intarea-provisioning-domains}}. When requesting
+the PvD information, the query and fragment components of the requested path SHOULD be left
```
[the remark this answers is not in the document]
Comment (tfpauly) [reply draft-pauly-adaptive-dns-privacy.md]: Reworded to just say it is left empty; it's not really a normative thing, it's just that I'm pointing out that the query string doesn't have the same meaning as DoH here.

**Pull request merged**